src/core/library.c

```diff
@@ -856,6 +856,9 @@ QuicLibrarySetGlobalParam(
 
         MsQuicLib.Settings.LoadBalancingMode = *(uint16_t*)Buffer;
         MsQuicLib.Settings.IsSet.LoadBalancingMode = TRUE;
+
+        QuicLibApplyLoadBalancingSetting();
```

Comment:
I think this function will need some synchronization if this function is called while in the middle of operation (i.e. connections exist).

Reply:
I read L848 that this set op will fail early if `MsQuicLib.InUse` is true, so I assume there is no ongoing traffic (No active registration) at the time of this applying?

Reply:
or do you mean we need atomic changes in `QuicLibApplyLoadBalancingSetting`?

Reply:
I think the simplest solution is to restrict this SetParam to only work when `!MsQuicLib.InUse`, and then there should be no race condition related to calling `QuicLibApplyLoadBalancingSetting`. Does that work for you?

Reply:
I am ok with this, I understand I cannot change LB mode once you have traffic running that means `MsQuicLib.InUse` is `false` and there is no registration. 
My understanding is we have it already, just few lines above.

https://github.com/microsoft/msquic/blob/8faef0db35c9d8d6758f0bf976de22bfc6b05951/src/core/library.c#L848-L854

`QuicLibApplyLoadBalancingSetting` will not be called if `MsQuicLib.InUse` is true.

So what I should update now is to protects these lines with mutex to avoid race of calling `QuicLibApplyLoadBalancingSetting` ?

Reply:
Ah, you're right. I missed that somehow. So, yeah, I think this PR is good to go then. Thanks for baring with me!